api/grpc/mpi/v1/command.proto

```diff
@@ -0,0 +1,358 @@
+// These proto definitions follow https://protobuf.dev/programming-guides/style/
+// and recommendations outlined in https://static.sched.com/hosted_files/kccncna17/ad/2017%20CloudNativeCon%20-%20Mod%20gRPC%20Services.pdf
+syntax = "proto3";
+package f5.nginx.agent.api.grpc.mpi.v1;
+
+option go_package = "github.com/nginx/agent/v3/api/grpc/mpi/v1";
+
+import "common.proto";
+import "file.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+// A service outlining the command and control options for a DataPlane
+service CommandService {
+    // Connects NGINX Agents to the Management Plane agnostic of instance data
+    rpc Connect(ConnectionRequest) returns (ConnectionResponse) {}
+    // Reports on instances and their configurations
+    rpc Status(DataPlaneStatus) returns (google.protobuf.Empty) {}
+    // Reports on instance health
+    rpc Health(DataPlaneHealth) returns (google.protobuf.Empty) {}
+    // A decoupled communication mechanism between the data plane and management plane.
+    rpc Subscribe(stream DataPlaneMessage) returns (stream ManagementPlaneMessage) {}
+}
+
+// The connection request is an intial handshake to establish a connection, sending NGINX Agent instance information
+message ConnectionRequest {
+    // Meta-information associated with a message
+    f5.nginx.agent.api.grpc.mpi.v1.common.MessageRequest message_metadata = 1;
+    // instance information associated with the NGINX Agent
+    Instance agent = 2;
+}
+
+// This contains a series of NGINX Agent configurations
+message AgentConfig { 
+    // Command server settings
+    Command command = 1;
+    // Metrics server settings
+    Metrics metrics = 2;
+    // A series of key/value pairs to add more data to the NGINX Agent instance
+    repeated google.protobuf.Struct labels = 3;
+    // A list of features that the NGINX Agent has
+    repeated string features = 4;
+    // Message buffer size, maximum not acknowledged messages from the subscribe perspective
+    string message_buffer_size = 5;
+    // 
+}
+
+// The command settings, associated with messaging from an external source 
+message Command {
+    // The connection and security settingss for the command server
+    ConnectionSettings connection_settings = 1;
+}
+
+// The metrics settings associated with orgins (sources) of the metrics and destinations (exporter)
+message Metrics {
```

Comment:
Metrics feels like a one off, not used outside of AgentConfig, consider embed.

Reply:
not sure how your link apply here

Reply:
Metrics and Command are wrappers for evolving proto specs and will diverge in the implementation. Burying them in the top level doesn't give us the flexibility to evolve